modules/standard/GMP.chpl

```diff
@@ -560,7 +560,7 @@ module GMP {
     }
     proc set_str(str:string, base:int=0)
     {
-      on this do mpz_set_str(this.mpz, str.c_str(), base.safeCast(c_int));
+      on this do mpz_set_str(this.mpz, str.localize().c_str(), base.safeCast(c_int));
```

Comment:
We were discussing a little bit about the pointers in file and channel, and I get the sense that there is some support for having a way to have a "wide pointer" that can be used for C interoperability. I think that would form an alternative to these .localize calls - or at least, improve error handling, since the compiler can add checks that the pointer is local as it is passed to any extern proc. It would form an alternative to .localize if we make the runtime understand what a "wide C string" is - but that seems like it would change a lot of code. Another option (maybe already brought up?) is to have the compiler add the .localize calls for you. (Ie, whenever you call .c_str, it would change it to .localize.c_str() ).

Another way to do it is for .c_str() to return a record storing a c_ptr and a locale and to support coercions from this record to c_ptr. For that, maybe my user-defined coercions patch would help. The idea would be that c_str() would always "localize" and by returning a new record it would avoid the memory leak (since that record would be destroyed just as the call to localize is).

I think that .localize() is a reasonable initial strategy. I'd like to try to do better since it seems that it might be hard for a user to understand when .localize() is needed (always?).